crates/brainfuck_prover/src/components/memory/table.rs

```diff
@@ -267,6 +274,104 @@ impl MemoryColumn {
     }
 }
 
+/// The interaction elements drawn for the extension column of the Memory component.
+///
+/// The logup protocol uses these elements to combine the values of the different
+/// registers of the main trace to create a random linear combination
+/// of them, and use it in the denominator of the fractions in the logup protocol.
+///
+/// There are 3 lookup elements in the Memory component, as only the 'real' registers
+/// are used: `clk`, `mp` and `mv`. `d` is used to eventually nullify the numerator.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct MemoryElements(LookupElements<{ MemoryColumn::count() - 1 }>);
+
+impl MemoryElements {
+    /// Provides dummy lookup elements.
+    pub fn dummy() -> Self {
+        Self(LookupElements::dummy())
+    }
+
+    /// Draw random elements from the Fiat-Shamir [`Channel`].
+    ///
+    /// These elements are randomly secured, and will be use
+    /// to generate the interaction trace with the logup protocol.
+    pub fn draw(channel: &mut impl Channel) -> Self {
+        Self(LookupElements::draw(channel))
+    }
+}
+
+impl<F: Clone, EF: RelationEFTraitBound<F>> Relation<F, EF> for MemoryElements {
+    /// Combine multiple values from a basefield (e.g. [`BaseField`])
+    /// and combine them to a value from an extension field (e.g. [`PackedSecureField`])
+    ///
+    /// This is used when computing the interaction values from the main trace values.
+    fn combine(&self, values: &[F]) -> EF {
+        values
+            .iter()
+            .zip(self.0.alpha_powers)
+            .fold(EF::zero(), |acc, (value, power)| acc + EF::from(power) * value.clone()) -
+            self.0.z.into()
+    }
+
+    /// Returns the name of the struct.
+    fn get_name(&self) -> &str {
```

Comment:
`get_name` & `get_size` are from the trait `Relation` from stwo repo, we must use them as is :/

We could always make a PR for this but I don't think it would be merged

Reply:
Aaaah yes true, then forget about my comment